src/neural/backends/network_onnx.cc

```diff
@@ -287,20 +507,115 @@ void OnnxComputation<DataType>::ComputeBlocking() {
+        network_->provider_ == OnnxProvider::CUDA) {
+      if (i == 0) {
+        ReportCUDAErrors(
+            cudaStreamWaitEvent(network_->upload_stream_,
+                                inputs_outputs_->inputs_processed_event_));
+      }
+      const char* src_masks =
+          static_cast<char*>(inputs_outputs_->input_tensor_data_);
+      char* dst_masks =
+          static_cast<char*>(inputs_outputs_->input_tensor_upload_device_);
+      src_masks += i * kInputPlanes * sizeof(uint64_t);
+      dst_masks += i * kInputPlanes * (sizeof(uint64_t) + sizeof(DataType));
+      ReportCUDAErrors(cudaMemcpyAsync(
+          dst_masks, src_masks, batch * kInputPlanes * sizeof(uint64_t),
+          cudaMemcpyHostToDevice, network_->upload_stream_));
+      char* src_values =
+          static_cast<char*>(inputs_outputs_->input_tensor_data_);
+      src_values += network_->max_batch_size_ * kInputPlanes * sizeof(uint64_t);
+      src_values += i * kInputPlanes * sizeof(DataType);
+      char* dst_values = dst_masks + batch * kInputPlanes * sizeof(uint64_t);
+      ReportCUDAErrors(cudaMemcpyAsync(
+          dst_values, src_values, batch * kInputPlanes * sizeof(DataType),
+          cudaMemcpyHostToDevice, network_->upload_stream_));
+      ReportCUDAErrors(cudaEventRecord(inputs_outputs_->inputs_uploaded_event_,
+                                       network_->upload_stream_));
+      ReportCUDAErrors(cudaStreamWaitEvent(
+          network_->compute_stream_, inputs_outputs_->inputs_uploaded_event_));
+      if (network_->fp16_) {
+        half* dst =
+            reinterpret_cast<half*>(inputs_outputs_->input_tensor_data_device_);
+        dst += i * kInputPlanes * 8 * 8;
+        cudnn_backend::expandPlanesOnnx(dst, dst_masks, batch * kInputPlanes,
+                                        network_->compute_stream_);
+      } else if (network_->bf16_) {
+        __nv_bfloat16* dst = reinterpret_cast<__nv_bfloat16*>(
+            inputs_outputs_->input_tensor_data_device_);
+        dst += i * kInputPlanes * 8 * 8;
+        cudnn_backend::expandPlanesOnnx(dst, dst_masks, batch * kInputPlanes,
+                                        network_->compute_stream_);
+      } else {
+        float* dst = reinterpret_cast<float*>(
+            inputs_outputs_->input_tensor_data_device_);
+        dst += i * kInputPlanes * 8 * 8;
+        cudnn_backend::expandPlanesOnnx(dst, dst_masks, batch * kInputPlanes,
+                                        network_->compute_stream_);
+      }
+
+      ReportCUDAErrors(cudaEventRecord(inputs_outputs_->inputs_processed_event_,
+                                       network_->upload_stream_));
+      if (i == 0) {
+        ReportCUDAErrors(
+            cudaStreamWaitEvent(network_->compute_stream_,
+                                inputs_outputs_->outputs_download_event_));
+      }
+      options.AddConfigEntry("disable_synchronize_execution_providers", "1");
+    } else {
+      binding.SynchronizeInputs();
```

Comment:
Shouldn't this be outside the `#if` block? Something like
```diff
@@ -560,10 +565,12 @@ void OnnxComputation<DataType>::ComputeBlocking() {
                                 inputs_outputs_->outputs_download_event_));
       }
       options.AddConfigEntry("disable_synchronize_execution_providers", "1");
-    } else {
+    } else
+#endif
+    {
       binding.SynchronizeInputs();
     }
-#endif
+
     network_->session_[step - 1].Run(options, binding);
 #ifdef USE_ONNX_CUDART
     if (network_->provider_ == OnnxProvider::TRT ||
```
Same for `SynchronizeOutputs()` a bit further down.